editoast/src/views/timetable/train_schedule.rs

```
@@ -590,6 +609,309 @@ async fn occupancy_blocks(
+    let infra = Infra::retrieve_real_or_fail(db_pool.get().await?, infra_id, || {
+        TrainScheduleError::InfraNotFound { infra_id }
+    })
+    .await?;
+
+    let train_schedules: Vec<models::TrainSchedule> =
+        models::TrainSchedule::retrieve_batch_or_fail(
+            &mut db_pool.get().await?,
+            train_schedule_ids,
+            |missing| TrainScheduleError::BatchTrainScheduleNotFound {
+                number: missing.len(),
+            },
+        )
+        .await?;
+
+    let operational_point = OperationalPointModel::retrieve_real_or_fail(
+        db_pool.get().await?,
+        (infra_id, op_id.clone()),
+        || TrainScheduleError::OperationalPointNotFound {
+            op_id: op_id.clone(),
+        },
+    )
+    .await?;
+
+    // Retrieve simulations
+    let simulations_result = train_simulation_batch(
+        &mut db_pool.get().await?,
+        valkey,
+        core_client,
+        &train_schedules,
+        &infra,
+        electrical_profile_set_id,
+    )
+    .await?;
+
+    let operational_point_track_offsets = operational_point.schema.track_offset();
+
+    let path_items = train_schedules
+        .iter()
+        .flat_map(|ts| ts.path.iter().map(|p| &p.location))
+        .collect::<Vec<_>>();
+
+    let path_item_cache =
+        PathItemCache::load(&mut db_pool.get().await?, infra_id, &path_items).await?;
+
+    for ((simulation, pathfinding), train_schedule) in
+        simulations_result.iter().zip(train_schedules)
+    {
+        // Get the track ranges from the pathfinding
+        let (track_ranges, path_item_positions) = match pathfinding {
+            PathfindingResult::Success(pathfinding_result_success) => (
+                &pathfinding_result_success.track_section_ranges,
+                &pathfinding_result_success.path_item_positions,
+            ),
+            PathfindingResult::Failure(_) => {
+                tracing::info!(train_schedule.id, "pathfinding failed");
```

Doesn't explain why it's ok for a pathfinding to fail.

Shouldn't the error be returned in the response, at the very least for explicability? Silent failures are rarely a good thing.

---

I'm not sure it makes sense for the endpoint to notify the user of all pathfinding or simulation failures, especially if they are unrelated to the operational_point_id provided in the input.
The main goal of this endpoint is to return track occupancy.
Should I really return all failures, even those not directly linked to the requested operational point?
We can discuss that if needed. 

Btw, I've put a comment. I don't know if it suits you, let me know.

---

I understand the goal, but as an API user it's weird to explicitly ask for a result for TS 1, 2 and 3, but only to be handed that result for 1 and 2. What happened to 3? The only way to get a glimpse of information would be to look at editoast logs.

My point being: *we were asked* to perform a computation on a resource, we did it, we got a result, let's return it instead of trashing it. It's up to the API user to ignore information they don't need in their context. But I feel it's hardly an assumption we can make at API level.

---

I completely agree with you, I think it could be interesting to send back the failure if there is failure. However, for the moment, it's something that hasn't been designed and that would require changes on the front end. I think we can do an enhancement for this issue if that suits you?

---

Fully agreed for another PR, especially since this is not the only place where we discard meaningful errors in responses.